newrelic-security-agent/src/main/java/com/newrelic/api/agent/security/Agent.java

```diff
@@ -657,4 +658,15 @@ public void retransformUninstrumentedClass(Class<?> classToRetransform) {
             NewRelic.getAgent().getLogger().log(Level.FINER, "Class ", classToRetransform, " already instrumented.");
         }
     }
+
+    @Override
+    public String decryptAndVerify(String encryptedData, String hashVerifier) {
+        String decryptedData = EncryptorUtils.decrypt(AgentInfo.getInstance().getLinkingMetadata().get(INRSettingsKey.NR_ENTITY_GUID), encryptedData);
+        if(EncryptorUtils.verifyHashData(hashVerifier, decryptedData)) {
```

Comment:
missing null check on decrypted data

Reply:
Changes provided in [commit-b2c0f283](https://github.com/newrelic/csec-java-agent/pull/207/commits/b2c0f283004a50eb7e5ac778523355e482f6d6a1)